Modules/_tkinter.c

```diff
@@ -3022,17 +3048,86 @@ Tkapp_WantObjects(PyObject *self, PyObject *args)
 /*[clinic input]
 _tkinter.tkapp.willdispatch
 
+Skips main loop wait polling until next call of mainloop finishes
+
+DEPRECATED
+
+call(), var_invoke(), and createcommand() will wait indefinitely for main loop
+Sets the internal dispatching flag regardless of mainloop dispatch state
+May lead to unexpected hangs or inconsistencies in dispatching
+
 [clinic start generated code]*/
 
 static PyObject *
 _tkinter_tkapp_willdispatch_impl(TkappObject *self)
-/*[clinic end generated code: output=0e3f46d244642155 input=d88f5970843d6dab]*/
+/*[clinic end generated code: output=0e3f46d244642155 input=7b46376275304a4a]*/
 {
+    if (PyErr_WarnEx(PyExc_DeprecationWarning,
+        "willdispatch() is deprecated; consider polling dispatch() instead.", 1))
+    {
+        return NULL;
+    }
     self->dispatching = 1;
 
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+_tkinter.tkapp.dispatching
+
+Returns the internal dispatching state
+
+Returns 1 if the mainloop is running, or 0 if the mainloop is not running.
+
+[clinic start generated code]*/
+
+static PyObject *
+_tkinter_tkapp_dispatching_impl(TkappObject *self)
+/*[clinic end generated code: output=1b0192766b008005 input=fca22ac098f764ff]*/
+{
+    return PyLong_FromLong(self->dispatching);
```

Comment:
```suggestion
    return PyBool_FromLong(self->dispatching);
```
I think we are better returning a bool than int